packages/lending/src/adapter.ts

```diff
@@ -174,78 +174,86 @@ export class Adapter extends common.Web3Toolkit {
       const destCollateral = portfolio.findSupply(destToken);
 
       if (srcCollateral && destCollateral) {
-        // 1. validate src amount
-        if (new BigNumberJS(srcAmount).gt(srcCollateral.balance)) {
+        // 1. validate withdraw
+        if (!srcCollateral.validateWithdraw(srcAmount)) {
           output.error = { name: 'srcAmount', code: 'INSUFFICIENT_AMOUNT' };
         }
         output.afterPortfolio.withdraw(srcCollateral.token, srcAmount);
```

Comment:
Is this expected that if there are any `output.error`,  the `output.afterPortfolio` will still need to be calculated?

Reply:
The current process involves updating the after-portfolio based on the user's input. If adjustments are necessary, we should discuss with the product manager whether specifications need to be modified. cc @zodahu

Reply:
The current implementation seems good as it allows users to preview their portfolio even if they don't have the funds. This enhances the user experience by enabling users to preview potential outcomes without needing to have the funds.

Reply:
OK, just to rephrase that: The current process is such that if there are any errors, it won't execute the code to build `destAmount` and `logics`. However, `afterPortfolio` will still be calculated regardless.

```typescript
export type OperationOutput = {
  destAmount: string;
  afterPortfolio: Portfolio;
  error?: OperationError;
  logics: apisdk.Logic[];
};
```